identity-wallet/src/state/profile_settings/reducers/update_sorting_preference.rs

```diff
@@ -58,45 +60,91 @@ pub async fn sort_credentials(state: AppState, _action: Action) -> Result<AppSta
-    let issuance_new_old =
-        |a: &DisplayCredential, b: &DisplayCredential| a.metadata.date_issued.cmp(&b.metadata.date_issued);
-    let added_new_old =
-        |a: &DisplayCredential, b: &DisplayCredential| a.metadata.date_added.cmp(&b.metadata.date_added);
+    let list: Vec<String> = credentials.iter().map(|x| x.display_name.clone()).collect();
+    let sorted_list = sort(list, state.profile_settings.locale.clone());
+
+    let name_az = |a: &DisplayCredential, b: &DisplayCredential| {
+        let pos_a = sorted_list
+            .iter()
+            .position(|display_name| display_name == &a.display_name)
+            .unwrap();
+        let pos_b = sorted_list
+            .iter()
+            .position(|display_name| display_name == &b.display_name)
+            .unwrap();
+        pos_a.cmp(&pos_b)
+    };
 
-    credentials.sort_by(match preferences.sort_method {
-        CredentialSortMethod::NameAZ => name_az,
-        CredentialSortMethod::IssueDateNewOld => issuance_new_old,
-        CredentialSortMethod::AddedDateNewOld => added_new_old,
-    });
+    match preferences.sort_method {
+        CredentialSortMethod::NameAZ => credentials.sort_by(name_az),
+        CredentialSortMethod::IssueDateNewOld => credentials.sort_by(|a: &DisplayCredential, b: &DisplayCredential| {
+            a.metadata.date_issued.cmp(&b.metadata.date_issued)
+        }),
+        CredentialSortMethod::AddedDateNewOld => credentials
+            .sort_by(|a: &DisplayCredential, b: &DisplayCredential| a.metadata.date_added.cmp(&b.metadata.date_added)),
+    };
 
     if preferences.reverse {
         credentials.reverse();
     }
 
+    // current_user_prompt is not set to None,
+    // as this reducer is often used in combination with reducers that need to send a user_prompt to the frontend.
     Ok(AppState { credentials, ..state })
 }
 
 pub async fn sort_connections(state: AppState, _action: Action) -> Result<AppState, AppError> {
     let mut connections: Vec<Connection> = state.connections.0.clone();
     let preferences: Preferences<ConnectionSortMethod> = state.profile_settings.sorting_preferences.connections.clone();
 
-    let name_az = |a: &Connection, b: &Connection| a.name.cmp(&b.name);
-    let first_interacted_new_old = |a: &Connection, b: &Connection| a.first_interacted.cmp(&b.first_interacted);
-    let last_interacted_new_old = |a: &Connection, b: &Connection| a.last_interacted.cmp(&b.last_interacted);
+    let list: Vec<String> = connections.iter().map(|x| x.name.clone()).collect();
+    let sorted_list = sort(list, state.profile_settings.locale.clone());
 
-    connections.sort_by(match preferences.sort_method {
-        ConnectionSortMethod::NameAZ => name_az,
-        ConnectionSortMethod::FirstInteractedNewOld => first_interacted_new_old,
-        ConnectionSortMethod::LastInteractedNewOld => last_interacted_new_old,
-    });
+    let name_az = |a: &Connection, b: &Connection| {
```

Comment:
you mean in the Credential function? because it is slightly different there since the enum the match case is based on is a different enum, so i don't really see a convenient way to abstract a general function here